noxfile.py

```diff
@@ -0,0 +1,261 @@
+Unlike tox, Nox uses a standard Python file for configuration.
+
+To run all sessions defined in the noxfile.py file, run the following command:
+$ nox
+
+To run a specific session, run the following command:
+$ nox -s <session-name>
+
+To run a session with additional arguments, run the following command:
+$ nox -s <session-name> -- <additional-arguments>
+
+To list all available sessions, run the following command:
+$ nox -l
+"""
+
+import nox
+from nox.sessions import Session
+
+# default sessions (sorted alphabetically)
+nox.options.sessions = ["bandit", "black", "flake8", "isort", "mypy", "unit-tests"]
+
+
+def _install_requirements(session: Session) -> None:
+    session.install("-r", "requirements-extras.txt")
+
+
+@nox.session(reuse_venv=True)
+def bandit(session: Session) -> None:
+    """Run bandit on cachi2 directory"""
+    _install_requirements(session)
+    session.run(
+        "bandit",
+        "-c",
+        "pyproject.toml",
+        "-r",
+        "cachi2",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def black(session: Session) -> None:
+    """Run black on cachi2 and tests directories and noxfile.py"""
+    _install_requirements(session)
+    session.run(
+        "black",
+        "--check",
+        "--diff",
+        "cachi2",
+        "tests",
+        "noxfile.py",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def flake8(session: Session) -> None:
+    """Run flake8 on cachi2 and tests directories"""
+    _install_requirements(session)
+    # TODO: Move flake8-pyproject to requirements-extras.txt
+    session.install("flake8-pyproject")
+    session.run(
+        "flake8",
+        "cachi2",
+        "tests",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def isort(session: Session) -> None:
+    """Run isort on cachi2 and tests directories and noxfile.py"""
+    _install_requirements(session)
+    session.run(
+        "isort",
+        "--check",
+        "--diff",
+        "--color",
+        "cachi2",
+        "tests",
+        "noxfile.py",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(reuse_venv=True)
+def mypy(session: Session) -> None:
+    """Run mypy on cachi2 and tests directories"""
+    _install_requirements(session)
+    session.run(
+        "mypy",
+        "--install-types",
+        "--non-interactive",
+        "cachi2",
+        "tests",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(name="unit-tests", reuse_venv=True, python=["3.9", "3.10", "3.11", "3.12"])
+def unit_tests(session: Session) -> None:
+    """Run unit tests and generate coverage report"""
+    _install_requirements(session)
+    session.run(
+        "pytest",
+        "--log-level=DEBUG",
+        "--cov=cachi2",
+        "--cov-config=pyproject.toml",
+        "--cov-report=term",
+        "--cov-report=html",
```

Comment:
While we are here could you please add `--no-cov-on-fail`? Coverage report is a bit useless when there are failures, but consumes screen space.